rootex/core/renderer/materials/textured_material.cpp

```diff
@@ -38,3 +57,57 @@ void TexturedMaterial::bind()
 	setPSConstantBuffer({ LightSystem::GetSingleton()->getLights() });
 	setPSConstantBuffer(PSDiffuseConstantBufferMaterial({ 0.6f, 30.0f, { 0.0f, 0.0f } }));
 }
+
+JSON::json TexturedMaterial::getJSON() const
+{
+	JSON::json j;
+
+	j["type"] = "Textured Material";
+	j["imageFile"] = m_ImageFile->getPath();
+
+	return j;
+}
+
+void TexturedMaterial::setTexture(ImageResourceFile* image)
+{
+	Ref<Texture> texture(new Texture(image));
+	m_ImageFile = image;
+	m_DiffuseTexture = texture;
+}
+
+#ifdef ROOTEX_EDITOR
+#include "imgui.h"
+#include "imgui_stdlib.h"
+void TexturedMaterial::draw()
+{
+	ImGui::Text("Textured Material");
+	m_ImagePathUI = m_ImageFile->getPath().string();
+	if (ImGui::InputText("Texture", &m_ImagePathUI, ImGuiInputTextFlags_EnterReturnsTrue))
+	{
+		ImageResourceFile* image = ResourceLoader::CreateImageResourceFile(m_ImagePathUI);
+		if (image)
+		{
+			setTexture(image);
+		}
+	}
+
+	if (ImGui::BeginDragDropTarget())
+	{
+		if (const ImGuiPayload* payload = ImGui::AcceptDragDropPayload("Resource Drop"))
+		{
+			const char* payloadFileName = (const char*)payload->Data;
+			FilePath payloadPath(payloadFileName);
+			FilePath ext = payloadPath.extension();
+			if (ext == ".jpg" || ext == ".png" || ext == ".jpeg")
```

Comment:
Remind me when you see this comment. I have made a solution that will not hardcode the extension names everywhere. I will have to do changes in this area when I rebase